### PR TITLE
bazel: update sql-parser

### PR DESCRIPTION
**Pull request by ipuustin**
Description: 

Update sql-parser to latest. The change fixes the Envoy compilation warning:

```
INFO: From Compiling external/com_github_envoyproxy_sqlparser/src/parser/flex_lexer.cpp:
flex_lexer.cpp:3577:16: warning: function 'yyinput' is not needed and will not be emitted [-Wunneeded-internal-declaration]
    static int yyinput (yyscan_t yyscanner)
               ^
1 warning generated.
```

Risk Level: low
Testing: mysql extension unit tests
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by dio** [comment]
@ipuustin thanks!

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -74,9 +74,9 @@ REPOSITORY_LOCATIONS = dict(
         urls = ["https://github.com/Cyan4973/xxHash/archive/v0.7.2.tar.gz"],
     ),
     com_github_envoyproxy_sqlparser = dict(
-        sha256 = "425dfee0c4fe9aff8acf2365cde3dd2ba7fb878d2ba37562d33920e34c40c05e",
-        strip_prefix = "sql-parser-5f50c68bdf5f107692bb027d1c568f67597f4d7f",
-        urls = ["https://github.com/envoyproxy/sql-parser/archive/5f50c68bdf5f107692bb027d1c568f67597f4d7f.tar.gz"],
+        sha256 = "a55c402ea04b76517ea4f3b9911b565738453208111ab9d64f5786dd21c6571b",
+        strip_prefix = "sql-parser-2db09ffd0d9740f4e6d5fdbfaec655e50b4154d4",
+        urls = ["https://github.com/ipuustin/sql-parser/archive/2db09ffd0d9740f4e6d5fdbfaec655e50b4154d4.tar.gz"],
```
Comment (dio) [inline bazel/repository_locations.bzl]: We will move this to sql-parser master once this is validated and https://github.com/envoyproxy/sql-parser/pull/2 get merged.

**Comment by dio** [comment]
@ipuustin https://github.com/envoyproxy/sql-parser/pull/2 is merged. Please update with the corresponding SHA. Thanks!

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -74,9 +74,9 @@ REPOSITORY_LOCATIONS = dict(
         urls = ["https://github.com/Cyan4973/xxHash/archive/v0.7.2.tar.gz"],
     ),
     com_github_envoyproxy_sqlparser = dict(
-        sha256 = "425dfee0c4fe9aff8acf2365cde3dd2ba7fb878d2ba37562d33920e34c40c05e",
-        strip_prefix = "sql-parser-5f50c68bdf5f107692bb027d1c568f67597f4d7f",
-        urls = ["https://github.com/envoyproxy/sql-parser/archive/5f50c68bdf5f107692bb027d1c568f67597f4d7f.tar.gz"],
+        sha256 = "8ccd2fe9087d96a91eda0929f3a0bdb5d9f877dd9cdc91f9fcf85385fe70cc14",
+        strip_prefix = "sql-parser-85904962d8723bbd3ae280dfb13cf4cc563faabc",
```
Comment (moderation) [inline bazel/repository_locations.bzl]: For consistency with other non-release dependencies, please add a comment line below here with:
```
# 2020-01-08
```

Comment (ipuustin) [reply bazel/repository_locations.bzl]: Sure, I'll add it.

**Pull request merged**